templates/tapestry.js

```diff
@@ -210,6 +210,23 @@ function tapestryTool(config){
                 { authenticated: ["read"] }
             );
             updatedNode.permissionsOrder = reorderPermissions(updatedNode.permissions);
+
+            if (node.mediaType === "accordion") {
+                const accordionProgress = []
+                const directChildren = this.dataset.links.filter(link => {
+                    return link.source == node.id
+                }).map(link => link.target)
+                directChildren.forEach(childId => {
+                    const child = this.dataset.nodes[findNodeIndex(childId)]
+                    child.presentationStyle = "accordion-row"
+                    
+                    if (child.completed) {
+                        accordionProgress.push(childId)
+                    }
+                })
+                updatedNode.accordionProgress = accordionProgress
+            }
```

Comment:
Progress update is done further up in the code using a different function that handles everything. Can you please put it there instead to keep the code clean and readable?

Reply:
Fixed, thanks